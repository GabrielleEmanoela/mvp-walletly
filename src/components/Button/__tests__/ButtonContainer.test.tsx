import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { ButtonContainer, ButtonProps } from '..';

const mock: ButtonProps = {
  children: 'Press Me',
  onPress: jest.fn(),
  loading: false,
  disabled: false,
  contentStyle: {},
};

describe('ButtonContainer', () => {
  it('should render correctly with all props', () => {
    const { getByText } = render(<ButtonContainer {...mock} />);
    expect(getByText('Press Me')).toBeOnTheScreen();
  });
  it('should call onPress when pressed', () => {
    const { getByText } = render(<ButtonContainer {...mock} />);
    fireEvent.press(getByText('Press Me'));
    expect(mock.onPress).toHaveBeenCalled();
  });
  it('should render correctly', () => {
    const { toJSON } = render(<ButtonContainer {...mock} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
