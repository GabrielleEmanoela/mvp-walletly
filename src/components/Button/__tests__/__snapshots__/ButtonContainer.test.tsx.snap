// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#4d539bff",
      "borderRadius": 40,
      "height": 55,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="ButtonContainer-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#4d539bff",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "flex": 1,
        "justifyContent": "center",
        "minWidth": 64,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="ButtonContainer-container"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 40,
          },
        ]
      }
      testID="ButtonContainer"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {},
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  {
                    "color": "rgba(255, 255, 255, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  },
                ],
              ],
            ]
          }
          testID="ButtonContainer-text"
        >
          Press Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;
